src/validation/forks.cpp

```diff
@@ -92,41 +82,57 @@ bool IsDAAEnabled(const Consensus::Params &consensusparams, const CBlockIndex *p
     return IsDAAEnabled(consensusparams, pindexTip->nHeight);
 }
 
-bool IsNov152018Scheduled() { return miningForkTime.Value() != 0; }
-bool IsNov152018Enabled(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
+bool IsNov152018Activated(const Consensus::Params &consensusparams, const int32_t nHeight)
+{
+    return nHeight >= consensusparams.nov2018Height;
+}
+
+bool IsNov152018Activated(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
 {
     if (pindexTip == nullptr)
     {
         return false;
     }
-    return pindexTip->IsforkActiveOnNextBlock(miningForkTime.Value());
+    return IsNov152018Activated(consensusparams, pindexTip->nHeight);
+}
+
+bool AreWeOnBCHChain()
+{
+    const Consensus::Params &consensusparams = Params().GetConsensus();
+    return ((miningForkTime.Value() != 0) && (IsNov152018Activated(consensusparams, chainActive.Tip())));
 }
 
-bool IsNov152018Next(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
+bool IsMay152019Enabled(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
 {
     if (pindexTip == nullptr)
     {
         return false;
     }
-    return pindexTip->forkAtNextBlock(miningForkTime.Value());
+    return pindexTip->IsforkActiveOnNextBlock(miningForkTime.Value());
 }
 
-
-bool IsSv2018Scheduled() { return miningSvForkTime.Value() != 0; }
-bool IsSv2018Enabled(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
+bool IsMay152019Next(const Consensus::Params &consensusparams, const CBlockIndex *pindexTip)
 {
     if (pindexTip == nullptr)
     {
         return false;
     }
-    return pindexTip->IsforkActiveOnNextBlock(miningSvForkTime.Value());
+    return pindexTip->forkAtNextBlock(miningForkTime.Value());
+}
+
+//* SV helpers/
+
+bool AreWeOnSVChain() { return miningSvForkTime.Value() != 0; }
```

Comment:
Suggest `assert(miningForkTime == 0)` here and the reverse in `AreWeOnBCHChain`. Don't use the DbgAssert, it's good to die when trying to enable incompatible consensus rules.

Reply:
`AreWeOnSVChain()` is meant to turn false if `miningForkTime != 0`, i.e. `miningSvForkTime == 0` rather than asserting i.e. crashing. 

The tweaks validator guarantee that if you can't set `miningForkTime` and `miningSvForkTime` both to something greater than `0`.

See line 232 and 256 of `src/unlimited.cpp`

So it seems impossible to me that `AreWeOnSVChain()` return true and at the same time we are going to apply BCH rule set.